src/macrofier/engine.mjs

```diff
@@ -486,29 +488,77 @@ function insertTableofContents(content) {
   return content
 }
 
+const isEnumType = x => x.type !== 'undefined' && x.type === 'string' && Array.isArray(x.enum)
+
+const getProperties = x => {
+   return Array.isArray(x.properties) ? x.properties[0] : x.properties
+}
+
+const isEnumProperties = schema => compose(
+    getProperties,
+    filter(enm => enm),
+    map(filter(enm => enm)),
+    map(props => props.map(([k, v]) => ((v.type === 'object') ? isEnumProperties(v) : ((v.type === 'array') ? isEnumType(v.items[0] ? v.items[0] : v.items): isEnumType(v))))),
+    map(Object.entries),
+    filter(schema => isObject(schema))
+)(schema)
+
+const getEnumProperties = schema => compose(
+    getProperties,
+    filter(enm => enm),
+    map(filter(isEnumType)),
+    map(props => props.map(([k, v]) => {
+      let enm = v
+      if (isEnumType(v) == true) {
+        enm = Object.assign({}, v)
+        enm.title = k
+      } else if (v.type === 'object') {
+        enm = getEnumProperties(v)
+      } else if (v.type === 'array') {
+        enm = Object.assign({}, (v.items[0] ? v.items[0] : v.items))
+        enm.title = k
+      }
+      return enm
+    })),
+    map(Object.entries),
+    filter(schema => isObject(schema))
+)(schema)
+
+const convertEnumTemplate = (sch, templateName, templates) => {
+  const template = getTemplate(templateName, templates).split('\n')
+  let schema = isEnumType(sch) ? sch : getEnumProperties(sch)
+  for (var i = 0; i < template.length; i++) {
+    if (template[i].indexOf('${key}') >= 0) {
+      template[i] = schema.enum.map(value => {
+        const safeName = value.split(':').pop().replace(/[\.\-]/g, '_').replace(/\+/g, '_plus').replace(/([a-z])([A-Z0-9])/g, '$1_$2').toUpperCase()
+        return template[i].replace(/\$\{key\}/g, safeName)
+                          .replace(/\$\{value\}/g, value)
+      }).join('\n')
+      if (!templateName.includes(".cpp")) {
+        template[i] = template[i].replace(/,*$/, '');
+      }
+    }
+  }
+  return template.join('\n')
+                 .replace(/\$\{title\}/g, capitalize(schema.title))
+                 .replace(/\$\{description\}/g, schema.description ? ('- ' + schema.description) : '')
+                 .replace(/\$\{name\}/g, schema.title)
+                 .replace(/\$\{NAME\}/g, schema.title.toUpperCase())
+}
+
 const enumFinder = compose(
-  filter(x => x.type === 'string' && Array.isArray(x.enum) && x.title),
+  filter(x => ((isEnumType(x) && x.title) || isEnumProperties(x))),
   map(([_, val]) => val),
   filter(([_key, val]) => isObject(val))
 )
 
-const generateEnums = (json, templates) => {
+const generateEnums = (json, templates, options = { destination: '' }) => {
+  const suffix = options.destination.split('.').pop()
   return compose(
     option(''),
+    map(val => val ? (suffix && getTemplate(`/sections/enum.${suffix}`, templates) ? getTemplate(`/sections/enum.${suffix}`, templates).replace(/\$\{schema.list\}/g, val.trimEnd()) : val) : ''),
```

Comment:
Having trouble understanding this line. Are you trying to dynamically swap which section template you use based on suffix? If so, you'll need to support cases where you don't know the suffix (i.e. just calling `getTemplate('/sections/enum')`.

It's probably easier to just always grab the suffix-less template for sections. We can add suffix-aware templates as needed to avoid the code getting overly complex.

Reply:
Oh, and there's no template in your PR for `/sections/enum.*` I think you forgot to add new files.